monix-reactive/shared/src/main/scala/monix/reactive/Observable.scala

```diff
@@ -607,6 +607,20 @@ abstract class Observable[+A] extends Serializable { self =>
   final def consumeWith[R](f: Consumer[A, R]): Task[R] =
     f(self)
 
+  /** Polymorphic version [[consumeWith]] that can work with generic
+    * `F[_]` tasks, anything that's supported via [[monix.eval.TaskLift]]
+    * conversions.
+    *
+    * So you can work among others with:
+    *
+    *  - `cats.effect.IO`
+    *  - `monix.eval.Coeval`
+    *  - `scala.concurrent.Future`
+    *  - ...
+    */
+  final def consumeWithF[R, F[_]](f: Consumer[A, R])(implicit F: TaskLift[F]): F[R] =
```

Comment:
Sorry for not noticing right away, the convention is to have `F[_]` first
Otherwise looks great :D